src/AlgoliaSearchCore.js

```diff
@@ -154,7 +154,9 @@ AlgoliaSearchCore.prototype.setExtraHeader = function(name, value) {
 * @param algoliaAgent the agent to add
 */
 AlgoliaSearchCore.prototype.addAlgoliaAgent = function(algoliaAgent) {
-  this._ua += ';' + algoliaAgent;
+  if (this._ua.indexOf(';' + algoliaAgent) === -1) {
```

Comment:
You're right. It's nicer but mostly useless; with an array you have to:
 - serialize the array at each request
 - finding if a UA is already set is marginally better in time (because we don't have much data)